File: app/src/main/java/com/charlie/sawl/schedule/Schedule.java
package com.charlie.sawl.schedule;

import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AlertDialog;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import com.charlie.sawl.R;
import java.util.ArrayList;
import java.util.Calendar;
import java.util.List;

public class Schedule extends Fragment {

    ViewPager mViewPager;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setHasOptionsMenu(true);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_tab, container, false);

        mViewPager = rootView.findViewById(R.id.mViewPager);
        if (mViewPager != null) {
            setupViewPager(mViewPager);
        }

        TabLayout tabLayout = rootView.findViewById(R.id.mTabLayout);
        tabLayout.setupWithViewPager(mViewPager);
        tabLayout.setTabMode(TabLayout.MODE_SCROLLABLE);
        setCurrentItem();

        return rootView;
    }

    private void setupViewPager(ViewPager viewPager) {
        Adapter mAdapter = new Adapter(getActivity().getSupportFragmentManager());

        mAdapter.addFragment("3월", ScheduleContents.getInstance(3));
        mAdapter.addFragment("4월", ScheduleContents.getInstance(4));
        mAdapter.addFragment("5월", ScheduleContents.getInstance(5));
        mAdapter.addFragment("6월", ScheduleContents.getInstance(6));
        mAdapter.addFragment("7월", ScheduleContents.getInstance(7));
        mAdapter.addFragment("8월", ScheduleContents.getInstance(8));
        mAdapter.addFragment("9월", ScheduleContents.getInstance(9));
        mAdapter.addFragment("10월", ScheduleContents.getInstance(10));
        mAdapter.addFragment("11월", ScheduleContents.getInstance(11));
        mAdapter.addFragment("12월", ScheduleContents.getInstance(12));
        mAdapter.addFragment("2019년 1월", ScheduleContents.getInstance(1));
        mAdapter.addFragment("2019년 2월", ScheduleContents.getInstance(2));

        viewPager.setAdapter(mAdapter);
    }

    @Override
    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        inflater.inflate(R.menu.menu_schedule, menu);
        super.onCreateOptionsMenu(menu, inflater);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.information:
                AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
                builder.setTitle("공지사항");
                builder.setMessage("1. 본 학사일정은 학교 홈페이지에 기초하여 작성되었으며, 일정은 학교 사정에 의해 언제든지 변경될 수 있습니다.\n" +
                        "\n" +
                        "2. 재학생들에게 도움이 될 만한 일정만 골라서 작성되었으므로 일부 일정은 빠져있습니다.\n" +
                        "\n" +
                        "3. 퇴사주 금요일만 본 학사일정에 작성하였습니다. 퇴사로 표시되어 있지 않은 날은 모두 잔류이며, 퇴사 일정에 대한 정확한 정보는 꼭 사감실에서 확인 바랍니다.");
                builder.setPositiveButton(android.R.string.ok, null);
                builder.show();
                break;
        }
        return true;
    }

    private void setCurrentItem() {
        int month = Calendar.getInstance().get(Calendar.MONTH);

        if (month >= 2) month -= 2;
        else month += 9;

        mViewPager.setCurrentItem(month);
    }

    private class Adapter extends FragmentStatePagerAdapter {
        private final List<Fragment> mFragments = new ArrayList<>();
        private final List<String> mFragmentTitles = new ArrayList<>();

        Adapter(FragmentManager manager) {
            super(manager);
        }

        void addFragment(String mTitle, Fragment mFragment) {
            mFragments.add(mFragment);
            mFragmentTitles.add(mTitle);
        }

        @Override
        public Fragment getItem(int position) {
            return mFragments.get(position);
        }

        @Override
        public int getCount() {
            return mFragments.size();
        }

        @Override
        public CharSequence getPageTitle(int position) {
            return mFragmentTitles.get(position);
        }
    }
}

File: app/src/main/kotlin/com/charlie/sawl/schedule/Schedule.kt
package com.charlie.sawl.schedule

import android.os.Bundle
import com.google.android.material.tabs.TabLayout
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import androidx.viewpager.widget.ViewPager
import androidx.appcompat.app.AlertDialog
import android.view.LayoutInflater
import android.view.Menu
import android.view.MenuInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import com.charlie.sawl.R
import java.util.ArrayList
import java.util.Calendar

class Schedule : Fragment() {

    private lateinit var mViewPager: ViewPager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setHasOptionsMenu(true)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val rootView = inflater.inflate(R.layout.fragment_tab, container, false)

        mViewPager = rootView.findViewById(R.id.mViewPager)
        setupViewPager(mViewPager)

        val tabLayout = rootView.findViewById<TabLayout>(R.id.mTabLayout)
        tabLayout.setupWithViewPager(mViewPager)
        tabLayout.tabMode = TabLayout.MODE_SCROLLABLE
        setCurrentItem()

        return rootView
    }

    private fun setupViewPager(viewPager: ViewPager) {
        val mAdapter = Adapter(activity!!.supportFragmentManager)

        mAdapter.addFragment(getText(R.string.month3).toString(), ScheduleContents.getInstance(3))
        mAdapter.addFragment(getText(R.string.month4).toString(), ScheduleContents.getInstance(4))
        mAdapter.addFragment(getText(R.string.month5).toString(), ScheduleContents.getInstance(5))
        mAdapter.addFragment(getText(R.string.month6).toString(), ScheduleContents.getInstance(6))
        mAdapter.addFragment(getText(R.string.month7).toString(), ScheduleContents.getInstance(7))
        mAdapter.addFragment(getText(R.string.month8).toString(), ScheduleContents.getInstance(8))
        mAdapter.addFragment(getText(R.string.month9).toString(), ScheduleContents.getInstance(9))
        mAdapter.addFragment(getText(R.string.month10).toString(), ScheduleContents.getInstance(10))
        mAdapter.addFragment(getText(R.string.month11).toString(), ScheduleContents.getInstance(11))
        mAdapter.addFragment(getText(R.string.month12).toString(), ScheduleContents.getInstance(12))
        mAdapter.addFragment(getText(R.string.month1).toString(), ScheduleContents.getInstance(1))
        mAdapter.addFragment(getText(R.string.month2).toString(), ScheduleContents.getInstance(2))

        viewPager.adapter = mAdapter
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        inflater.inflate(R.menu.menu_schedule, menu)
        super.onCreateOptionsMenu(menu, inflater)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.information -> {
                val builder = AlertDialog.Builder(activity!!)
                builder.setTitle(R.string.info)
                builder.setMessage(R.string.dialog_text)
                builder.setPositiveButton(android.R.string.ok, null)
                builder.show()
            }
        }
        return true
    }

    private fun setCurrentItem() {
        var month = 0

        val mCalendar = Calendar.getInstance()
        when {
            mCalendar.get(Calendar.MONTH) == Calendar.JANUARY -> month = 10
            mCalendar.get(Calendar.MONTH) == Calendar.FEBRUARY -> month = 11
            mCalendar.get(Calendar.MONTH) == Calendar.MARCH -> month = 0
            mCalendar.get(Calendar.MONTH) == Calendar.APRIL -> month = 1
            mCalendar.get(Calendar.MONTH) == Calendar.MAY -> month = 2
            mCalendar.get(Calendar.MONTH) == Calendar.JUNE -> month = 3
            mCalendar.get(Calendar.MONTH) == Calendar.JULY -> month = 4
            mCalendar.get(Calendar.MONTH) == Calendar.AUGUST -> month = 5
            mCalendar.get(Calendar.MONTH) == Calendar.SEPTEMBER -> month = 6
            mCalendar.get(Calendar.MONTH) == Calendar.OCTOBER -> month = 7
            mCalendar.get(Calendar.MONTH) == Calendar.NOVEMBER -> month = 8
            mCalendar.get(Calendar.MONTH) == Calendar.DECEMBER -> month = 9
        }

        mViewPager.currentItem = month
    }

    private inner class Adapter internal constructor(manager: FragmentManager) : FragmentStatePagerAdapter(manager) {
        private val mFragments = ArrayList<Fragment>()
        private val mFragmentTitles = ArrayList<String>()

        internal fun addFragment(mTitle: String, mFragment: Fragment) {
            mFragments.add(mFragment)
            mFragmentTitles.add(mTitle)
        }

        override fun getItem(position: Int): Fragment {
            return mFragments[position]
        }

        override fun getCount(): Int {
            return mFragments.size
        }

        override fun getPageTitle(position: Int): CharSequence? {
            return mFragmentTitles[position]
        }
    }
}